api/src/main/java/org/openmrs/module/queue/processor/VisitAttributeQueueNumberGenerator.java

```diff
@@ -0,0 +1,87 @@
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.queue.processor;
+
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.Date;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang.BooleanUtils;
+import org.apache.commons.lang.StringUtils;
+import org.openmrs.VisitAttribute;
+import org.openmrs.VisitAttributeType;
+import org.openmrs.api.APIException;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.VisitService;
+import org.openmrs.module.queue.model.Queue;
+import org.openmrs.module.queue.model.QueueEntry;
+import org.openmrs.module.queue.model.VisitQueueEntry;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * Implementation of a VisitQueueEntryProcessor which aims to retain legacy functionality using
+ * visit attributes
+ */
+@Slf4j
+@Component
+public class VisitAttributeQueueNumberGenerator implements VisitQueueEntryProcessor {
+	
+	@Autowired
+	VisitService visitService;
+	
+	@Autowired
+	AdministrationService adminService;
+	
+	/**
+	 * This populates the given VisitQueueEntry with an appropriate patient queue number
+	 */
+	public void beforeSaveVisitQueueEntry(VisitQueueEntry visitQueueEntry) {
+		QueueEntry queueEntry = visitQueueEntry.getQueueEntry();
+		Queue queue = queueEntry.getQueue();
+		VisitAttributeType visitAttributeType = getVisitAttributeType();
+		
+		LocalDate minDate = LocalDate.now();
+		LocalDate maxDate = LocalDate.now().plusDays(1);
+		Date startOfDay = Date.from(minDate.atStartOfDay(ZoneId.systemDefault()).toInstant());
+		Date endOfDay = Date.from(maxDate.atStartOfDay(ZoneId.systemDefault()).toInstant());
+		int visitQueueNumber = 1;
+		for (QueueEntry qe : queue.getQueueEntries()) {
+			if (BooleanUtils.isNotTrue(qe.getVoided())) {
+				if (!qe.getStartedAt().before(startOfDay) && qe.getStartedAt().before(endOfDay)) {
+					visitQueueNumber++;
+				}
+			}
+		}
+		String paddedString = StringUtils.leftPad(String.valueOf(visitQueueNumber), 3, "0");
+		String serviceName = queue.getName().toUpperCase();
+		String prefix = serviceName.length() < 3 ? serviceName : serviceName.substring(0, 3);
+		String queueNumber = prefix + "-" + paddedString;
+		
+		// Associate this queue number directly
+		queueEntry.setPatientQueueNumber(queueNumber);
+		
+		// Create Visit Attribute using generated queue number
+		VisitAttribute visitAttribute = new VisitAttribute();
+		visitAttribute.setAttributeType(visitAttributeType);
+		visitAttribute.setValueReferenceInternal(queueNumber);
+		visitQueueEntry.getVisit().addAttribute(visitAttribute);
```

Comment:
It's possible / likely that with the new column, this is not longer behavior that is desired.  But I'll leave that to others to discuss and decide.